### ISSUE TITLE
`array-bracket-spacing`: spread operator impacts `arraysInArrays` option

### ISSUE THREAD

**Issue by nwhittaker** [eslint-stylistic/eslint-stylistic#787]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Given the `array-bracket-spacing` rule configured always allow spaces except for arrays-in-arrays: Arrays containing spread arrays do not collapse the space between the closing brackets.

| Observed | Expected |
| - | - |
| `[ ...[ 1, 2, 3 ] ]` | `[ ...[ 1, 2, 3 ]]` |
| `[ 1, ...[ 2, 3 ] ]` | `[ 1, ...[ 2, 3 ]]` |
| `[ ...a ?? [ 1, 2, 3 ] ]` | `[ ...a ?? [ 1, 2, 3 ]]` |

In the case where the array is empty, I would be fine with keeping the space uncollapsed:
```js
[ ...[] ];
[ ...a ?? [] ];
```

### Reproduction

https://stackblitz.com/edit/github-i33ghmht

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by 9romise** [comment]
I do not fully understand the reason for doing this. Could you please explain? Or is it a more preferred style?
